### PR TITLE
Add modify

### PR DESCRIPTION
**Pull request by guilhermebodin**
This should solve #90 but tests are failing for different solvers and I don't know if this is something I can handle inside Dualization or the solver needs some different implementations.

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -89,6 +89,16 @@ function MOI.supports_constraint(
     return MOI.supports_add_constrained_variables(optimizer.dual_problem.dual_model, D)
 end
 
+function MOI.modify(optimizer::DualOptimizer{T},
+                    obj::MOI.ObjectiveFunction{MOI.ScalarAffineFunction{T}},
+                    obj_change::MOI.ScalarCoefficientChange{T}) where T
+    # We must find the constraint corresponding to the variable in the objective
+    # function and chaange its coeeficient on the constraint.
```
Comment (blegat) [inline src/MOI_wrapper.jl:96]: chaange -> change

**Review by blegat — COMMENTED** [review]
It's expected that Dualization fails if the underlying solver does not support modification. Is that the issue you're mentioning?

**Comment by guilhermebodin** [comment]
Yes, but I am not sure if, for example, GLPK supports modifications. Dualize, optimize, change parameters on the primal problem and optimize again is failing for GLPK. 
```julia
In `MathOptInterface.ScalarAffineFunction{Float64}`-in-`MathOptInterface.EqualTo{Float64}` constraint: Constant 5.0 of the function is not zero. The function constant should be moved to the set. You can use `MOI.Utilities.normalize_and_add_constraint` which does this automatically.
```

**Comment by blegat** [comment]
This error makes sense, the constant of the function should be zero, not sure, the set should be modified instead probably.

**Comment by guilhermebodin** [comment]
I found it a little tricky to understand that to modify the set I should use the `MOI.transform` function

**Comment by guilhermebodin** [comment]
@blegat

**Comment by blegat** [comment]
There are still failures in the tests, do you know where that is coming from ? You can replace 1.1, 1.2, 1.3 and 1.4 by 1 on travis.

File: src/MOI_wrapper.jl [hunk 2]
```diff
@@ -89,6 +89,20 @@ function MOI.supports_constraint(
     return MOI.supports_add_constrained_variables(optimizer.dual_problem.dual_model, D)
 end
 
+function MOI.modify(optimizer::DualOptimizer{T},
+                    ::MOI.ObjectiveFunction{MOI.ScalarAffineFunction{T}},
+                    obj_change::MOI.ScalarCoefficientChange{T}) where T
+    # We must find the constraint corresponding to the variable in the objective
+    # function and change its coeeficient on the constraint.
```
Comment (blegat) [inline src/MOI_wrapper.jl:96]: coeeficient -> coefficient

File: src/MOI_wrapper.jl [hunk 3]
```diff
@@ -89,6 +89,20 @@ function MOI.supports_constraint(
...
+    ci_to_change = optimizer.dual_problem.primal_dual_map.primal_var_dual_con[obj_change.variable]
+    sense_change = MOI.get(optimizer.dual_problem.dual_model, 
+                           MOI.ObjectiveSense()) == MOI.MAX_SENSE ? one(T) : -one(T)
+    MOI.transform(optimizer.dual_problem.dual_model,
```
Comment (blegat) [inline src/MOI_wrapper.jl:100]: What was the type of the constraint before ? It was not `EqualTo` ?

Comment (blegat) [reply src/MOI_wrapper.jl:100]: If it was `EqualTo` already, you can do `MOI.set(optimizer.dual_problem.dual_model, MOI.ConstraintSet(), ci_to_change, MOI.EqualTo(sense_change * obj_change.new_coefficient))`

**Comment by guilhermebodin** [comment]
@blegat Some compatibility issues, i tried to add Compat but no success, don`t know how to solve the problem yet
```
ERROR: Unsatisfiable requirements detected for package SemidefiniteModels [169818f4]:
 SemidefiniteModels [169818f4] log:
 ├─possible versions are: 0.1.0-0.1.1 or uninstalled
 ├─restricted by compatibility requirements with Compat [34da2185] to versions: uninstalled
 │ └─Compat [34da2185] log:
 │   ├─possible versions are: [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0-2.2.1, 3.0.0, 3.1.0, 3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0, 3.9.0-3.9.1, 3.10.0, 3.11.0, 3.12.0, 3.13.0, 3.14.0, 3.15.0, 3.16.0] or uninstalled
 │   └─restricted to versions 3.16.0 by an explicit requirement, leaving only versions 3.16.0
 └─restricted by compatibility requirements with CSDP [0a46da34] to versions: 0.1.1 — no versions left
   └─CSDP [0a46da34] log:
     ├─possible versions are: [0.3.0, 0.4.0-0.4.1, 0.5.0-0.5.4, 0.6.0-0.6.1] or uninstalled
     ├─restricted to versions * by an explicit requirement, leaving only versions [0.3.0, 0.4.0-0.4.1, 0.5.0-0.5.4, 0.6.0-0.6.1]
     ├─restricted by compatibility requirements with MathOptInterface [b8f27783] to versions: [0.4.1, 0.5.0-0.5.4, 0.6.0-0.6.1] or uninstalled, leaving only versions: [0.4.1, 0.5.0-0.5.4, 0.6.0-0.6.1]
     │ └─MathOptInterface [b8f27783] log:
     │   ├─possible versions are: [0.5.0-0.5.1, 0.6.0-0.6.4, 0.7.0, 0.8.0-0.8.4, 0.9.0-0.9.17] or uninstalled
     │   ├─restricted to versions 0.9.0-0.9 by Dualization [191a621a], leaving only versions 0.9.0-0.9.17
     │   │ └─Dualization [191a621a] log:
     │   │   ├─possible versions are: 0.3.3 or uninstalled
     │   │   └─Dualization [191a621a] is fixed to version 0.3.3
     │   └─restricted to versions 0.9.16 by an explicit requirement, leaving only versions 0.9.16
     └─restricted by compatibility requirements with Compat [34da2185] to versions: [0.5.0-0.5.4, 0.6.0-0.6.1] or uninstalled, leaving only versions: [0.5.0-0.5.4, 0.6.0-0.6.1]
       └─Compat [34da2185] log: see above
```

**Comment by guilhermebodin** [comment]
Crazy error 
```
glp_eval_tab_col: k = 6; variable must be non-basicError detected in file draft/glpapi12.c at line 936
signal (6): Aborted
```

**Comment by guilhermebodin** [comment]
replaced by #100

**Pull request closed without merging**